src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -234,6 +337,30 @@ public static TimeSpan FromMilliseconds(double value)
             return Interval(value, TicksPerMillisecond);
         }
 
+        /// <summary>
+        /// Returns a <see cref="TimeSpan"/> that represents a specified number of microseconds.
+        /// </summary>
+        /// <param name="value">A number of microseconds.</param>
+        /// <returns>An object that represents <paramref name="value"/>.</returns>
+        /// <exception cref="OverflowException">
+        /// <paramref name="value"/> is less than <see cref="MinValue"/> or greater than <see cref="MaxValue"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="value"/> is <see cref="double.PositiveInfinity"/>
+        ///
+        /// -or-
+        ///
+        /// <paramref name="value"/> is <see cref="double.NegativeInfinity"/>
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="value"/> is equal to <see cref="double.NaN"/>.
+        /// </exception>
+        public static TimeSpan FromMicroseconds(double value)
+        {
+            return Interval(value, TicksPerMicrosecond);
```

Comment:
> return Interval(value, TicksPerMicrosecond); [](http://example.com/codeflow?start=11&length=45)

Although this is similar to what we already have like FromSeconds and other methods, but in general I think the implementation of Interval method is wrong as it can overflow :-(

You don't have to change anything here but we may track fixing this with the issue [https://github.com/dotnet/runtime/issues/66815](https://github.com/dotnet/runtime/issues/66815)

Reply:
I added a comment